qiskit/primitives/base_estimator.py

```diff
@@ -105,13 +105,18 @@
 
 from abc import ABC, abstractmethod
 from collections.abc import Iterable, Sequence
+from copy import copy
```

Comment:
Can we update the example in this doc string to use names for the circuits and also show passing circuits and observables in some calls?

Reply:
I think it is important that both integers and objects can be used. Objects are easier to read, while integers are more important for performance. So, how about adding a note?

Reply:
Added in 5615215